### PR TITLE
Add option to use all columns for DRF tree splits

### PR DESCRIPTION
**Pull request by navdeep-G**

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/drf/DRF.java [hunk 1]
```diff
@@ -105,9 +105,15 @@ private void initTreeMeasurements() {
     @Override protected void initializeModelSpecifics() {
       _mtry_per_tree = Math.max(1, (int)(_parms._col_sample_rate_per_tree * _ncols));
       if (!(1 <= _mtry_per_tree && _mtry_per_tree <= _ncols)) throw new IllegalArgumentException("Computed mtry_per_tree should be in interval <1,"+_ncols+"> but it is " + _mtry_per_tree);
-      _mtry = (_parms._mtries==-1) ? // classification: mtry=sqrt(_ncols), regression: mtry=_ncols/3
-              ( isClassifier() ? Math.max((int)Math.sqrt(_ncols),1) : Math.max(_ncols/3,1))  : _parms._mtries;
-      if (!(1 <= _mtry && _mtry <= _ncols)) throw new IllegalArgumentException("Computed mtry should be in interval <1,"+_ncols+"> but it is " + _mtry);
+      if(_parms._mtries==-2){
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/tree/drf/DRF.java]: Is this going to work?

There is a this parameter validation check on DRF model builder:

  @Override public void init(boolean expensive) {
    super.init(expensive);
    // Initialize local variables
    if( _parms._mtries < 1 && _parms._mtries != -1 )
      error("_mtries", "mtries must be -1 (converted to sqrt(features)), or >= 1 but it is " + _parms._mtries);

File: h2o-algos/src/main/java/hex/tree/drf/DRF.java [hunk 2]
```diff
@@ -105,9 +105,15 @@ private void initTreeMeasurements() {
     @Override protected void initializeModelSpecifics() {
       _mtry_per_tree = Math.max(1, (int)(_parms._col_sample_rate_per_tree * _ncols));
       if (!(1 <= _mtry_per_tree && _mtry_per_tree <= _ncols)) throw new IllegalArgumentException("Computed mtry_per_tree should be in interval <1,"+_ncols+"> but it is " + _mtry_per_tree);
-      _mtry = (_parms._mtries==-1) ? // classification: mtry=sqrt(_ncols), regression: mtry=_ncols/3
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/tree/drf/DRF.java]: Please keep  the comment and describe what is mtries == -2 for

Comment (navdeep-G) [inline h2o-algos/src/main/java/hex/tree/drf/DRF.java]: Will add it back.

File: h2o-algos/src/main/java/hex/tree/drf/DRF.java (hunk 1 above)
Comment (navdeep-G) [inline h2o-algos/src/main/java/hex/tree/drf/DRF.java]: Oops, missed that. Thx

**Pull request merged**